src/camera.cc

```diff
@@ -118,10 +128,11 @@ ColorDbl Camera::Shade(Ray& ray, IntersectionPoint& p, Scene& scene) {
     Ray reflection_ray = Ray(reflection_point_origin, reflection_direction);
     // return p.get_material().get_specular() * Raytrace(reflection_ray, scene);
     return p.get_material().get_specular() * Raytrace(reflection_ray, scene);
-  //} else if (p.get_material().get_transparence() > 0.f) {
-  //  return COLOR_PINK;
   }
-
+  // If object has refractive component
+  if (p.get_material().get_transparence() > 0.f) {
```

Comment:
But yeah, the total inner reflection should probably call a function as it works exactly like the current perfect mirror reflectance.